x/application/keeper/transfer_applications.go

```diff
@@ -0,0 +1,241 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/pokt-network/poktroll/telemetry"
+	"github.com/pokt-network/poktroll/x/application/types"
+	"github.com/pokt-network/poktroll/x/shared"
+)
+
+// EndBlockerTransferApplication completes pending application transfers.
+// This always happens on the last block of a session during which the transfer started.
+// It is accomplished by:
+//  1. Copying the current state of the source app onto a new destination app
+//  2. Unstaking (removing) the source app
+//  3. Staking (storing) the destination app
+func (k Keeper) EndBlockerTransferApplication(ctx context.Context) error {
+	isSuccessful := false
+	defer telemetry.EventSuccessCounter(
+		"transfer_application_end",
+		telemetry.DefaultCounterFn,
+		func() bool { return isSuccessful },
+	)
+
+	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
+	currentHeight := sdkCtx.BlockHeight()
+	sharedParams := k.sharedKeeper.GetParams(ctx)
+	logger := k.Logger().With("method", "EndBlockerTransferApplication")
+
+	// Only process application transfers at the end of the session in
+	// order to avoid inconsistent/unpredictable mid-session behavior.
+	if !shared.IsSessionEndHeight(&sharedParams, currentHeight) {
```

Comment:
Add a debug log please that includes current height and the next session end height

Reply:
I would like to push back on this. **This would produce a log for every block that's not a session end height.** [`EndBlockerUnbondApplications()`]() and [`EndBlockerUnbondSuppliers()`]() both have a simliar conditional branch and neither includes the log that you're suggesting here.

Can you elaborate on the rationale or concern?

Reply:
I've added the current and session end height to the logger output:

```go
	currentHeight := sdkCtx.BlockHeight()
	sharedParams := k.sharedKeeper.GetParams(ctx)
	sessionEndHeight := shared.GetSessionEndHeight(&sharedParams, currentHeight)
	logger := k.Logger().
		With("method", "EndBlockerTransferApplication").
		With("current_height", currentHeight).
		With("session_end_height", sessionEndHeight)
```